contracts/gateway/Gateway.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../factory/IdFactory.sol";
+
+using ECDSA for bytes32;
+
+/// A required parameter was set to the Zero address.
+error ZeroAddress();
+/// The maximum number of signers was reached at deployment.
+error TooManySigners();
+/// The signed attempted to add was already approved.
+error SignerAlreadyApproved();
```

Comment:
maybe would make sense to add info about context into errors, in this case for example `error SignerAlreadyApproved()` could become `error SignerAlreadyApproved(address signer)` to make it easier for displaying the reason of errors on dapps, without needing to check back what was the parameters of the function call that failed.